### PR TITLE
feat: added Router and Header component

### PR DESCRIPTION
**Pull request by nghaosiong98**
* added react-router-dom package
* added Header
* added Header related buttons
* added Header related typographies
* added Header related inputs
* added Mui theming
* added custom breakpoint hook
* added plain Menu screen
* added Router

### CODE REVIEW

File: web/src/components/Input/index.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    height: "100%",
+    position: "absolute",
+    pointerEvents: "none",
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+}))
+
+export const SearchBar: FC = () => {
```
Comment (lee1409) [inline web/src/components/Input/index.tsx:40]: I think this lacks accepting props for search... follow the component name looks better

File: web/src/components/Input/index.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        classes={{
+          root: classes.inputRoot,
+          input: classes.inputInput,
+        }}
+        inputProps={{ "aria-label": "search" }}
+        placeholder="Search"
+      />
+    </div>
+  )
+}
```
Comment (lee1409) [inline web/src/components/Input/index.tsx:57]: Suggest rename it to SearchBar filename

File: web/src/components/Header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+              root: classes.toolBarRoot,
+            }}
+            disableGutters
+          >
+            <HeaderLogo>Owne</HeaderLogo>
+            <div className={classes.navigationBar}>
+              <HeaderNavButton>Explore</HeaderNavButton>
+              <HeaderNavButton>Menu</HeaderNavButton>
+              <HeaderNavButton>Stories</HeaderNavButton>
+            </div>
```
Comment (lee1409) [inline web/src/components/Header/index.tsx:60]: Try to abstract away from header, probably create a HeaderNavList or something

File: web/src/components/Header/index.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            <div className={classes.searchBar}>
+              <SearchBar />
+            </div>
```
Comment (lee1409) [inline web/src/components/Header/index.tsx:63]: If possible, fit the style by appending classname into searchBar will make the code looks neater. You can use clsx https://www.npmjs.com/package/clsx

**Review by lee1409 — COMMENTED** [review]
Reviewed :D

File: web/src/components/Input/index.tsx (hunk 1 above)
Comment (nghaosiong98) [reply web/src/components/Input/index.tsx:40]: ok

Comment (nghaosiong98) [reply web/src/components/Input/index.tsx:40]: the props not yet defined, lets add from time to time

File: web/src/components/Header/index.tsx (hunk 3 above)
Comment (nghaosiong98) [reply web/src/components/Header/index.tsx:60]: ok

File: web/src/components/Header/index.tsx (hunk 4 above)
Comment (nghaosiong98) [reply web/src/components/Header/index.tsx:63]: You means the className should be passed to SearchBar as a prop?

File: web/src/components/Typography/index.tsx [hunk 5]
```diff
@@ -1,4 +1,15 @@
-import React from "react"
 import { Typography } from "@material-ui/core"
 
 export const H1 = ({ ...props }) => <Typography variant="h1" {...props} />
+
+export const HeaderLogo = ({ ...props }) => (
```
[the remark this answers is not in the document]
Comment (nghaosiong98) [reply web/src/components/Typography/index.tsx:5]: Orite

File: web/src/components/Button/index.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  <IconButton {...props}>
+    <Search />
+  </IconButton>
+)
+
+export const AddButton = ({ ...props }: IconButtonProps) => (
+  <IconButton {...props}>
+    <AddCircleOutlineOutlined />
+  </IconButton>
+)
```
[the remark this answers is not in the document]
Comment (nghaosiong98) [reply web/src/components/Button/index.tsx:27]: Okay. I will do the checking with TS.

**Pull request merged**